Some small updates to the landing page (#33)
hpyproject/hpyproject.org: pages/index.rst

  implementation details are hidden.

.. _PyPy: https://pypy.org
.. _GraalPy:  https://www.graalvm.org/python
.. _`debug mode`: https://docs.hpyproject.org/en/latest/debug-mode.html
.. _`HPy Universal ABI`: https://docs.hpyproject.org/en/latest/overview.html#target-abis
.. _`PEP 620`: https://peps.python.org/pep-0620/

Current status
==============

HPy is under active development. `0.0.4`__ is the latest alpha release but we

> will soon leave the _alpha_ state and working hard towards a stable release.

Reviewer: ```suggestion
will soon leave the _alpha_ state and are working hard towards a stable release.
```

.. __: https://hpyproject.org/blog/posts/2022/06/hpy-0.0.4-third-public-release/

> In the last year, we invested a lot of effort in adding more APIs by migration

Reviewer: ```suggestion
We feel that the HPy ABI is stable enough now that we can fulfill our backwards and forwards binary compatibility promises with the upcoming release, and that the API now covers enough use cases to migrate
```

> important packages (in particular, checkout our numpy_ port). We also improved

Reviewer: ```suggestion
important packages (in particular, checkout our numpy_ port).
```

> our documentation a lot (in particular the `API reference`_).

Reviewer: ```suggestion
We also provide porting guides and extensive documentation (in particular the `API reference`_).
```

  course).

- numpy_: one of our ambitious goals is to port numpy to HPy, and to use this

>   experience to better understand how to design the API. This port are close to

Reviewer: ```suggestion
  experience to better understand how to design the API. This port is close to
```

  course).

- numpy_: one of our ambitious goals is to port numpy to HPy, and to use this
  experience to better understand how to design the API. This port is close to

>   pass the test suite.

Reviewer 2: ```suggestion
  passing the test suite.
```

.. _`kiwi-solver`: https://github.com/hpyproject/kiwi-hpy/
.. _`GraalVM blog post`: https://medium.com/graalvm/porting-matplotlib-from-c-api-to-hpy-aa32faa1f0b5

Where we need help
==================

> * **Documentation:** Our resources are very limied and we therefore concentrate

Reviewer 3: ```suggestion
* **Documentation:** Our resources are very limited and we therefore concentrate
```

Author: thanks for the catch

  on technical tasks. We already wrote a significant amount of documentation
  (see `Documentation`_) but it is not complete yet. If this non-coding work is
  something you can do, let us know.

* **Publicity:** HPy is already well received among core developers of some
  important projects. For instance, we are actively talking to Python, NumPy,
  and Cython core developers. However, we need to get more package developers to
  be interested in and even use HPy. We appreciate any help for advertising HPy
  and we are also happy to help preparing talks, demos, etc.

> * **Tooling:** This is  also a topic we would need help. We are, for example,

Reviewer 3: ```suggestion
* **Tooling:** This is also a topic we would need help. We are, for example,
```
